### ISSUE TITLE
Question: shell_environment:set_shell_var can i delete a variable

### ISSUE THREAD

**Issue by spbrogan** [tianocore/edk2-pytool-extensions#132]
Can i use None or "" to delete a variable?

**Comment by spbrogan** [comment]
def set_shell_var(self, var_name, var_data): in shell_environmanet.py:218

No function docstring for var_data or var_name

**Comment by matthewfcarlson** [comment]
I think this is related to #98 in general since we need to start adding docstrings and then add a way for them to be exposed

**Comment by Javagedes** [comment]
@spbrogan No, you cannot use None. You can slightly use "" as it will successfully replace the value with "".

Here are the results:

for `os.environ`
- None: Runtime error
- "": Replaces entry value with "".

for `self.active_environ`
- None: Replaces entry value with None
- "": Replaces entry value with "".

If you want this functionality, I suggest we pop the entry. I can do this in the same function by checking if the value is "" or None and then popping the entry, or I can create new functions. Do you have a preference?

**Comment by Javagedes** [comment]
Per external conversation, update set_shell_var to accept None to to delete the variable.

Additionally, deduplicate (remove from current location and insert at front) when using set_shell_var

**Comment by Javagedes** [comment]
Allowing users to delete shell / build vars nolonger makes sense via None as this conflicts with the non-valued build variable interface. Should we want to delete vars, which I don't know why we would in  the first place, we need to create new functions. I've updated the documentation via #432 to show this.
